samcli/commands/package/command.py

```diff
@@ -0,0 +1,144 @@
+"""
+CLI command for "package" command
+"""
+from functools import partial
+
+import click
+
+from samcli.cli.main import pass_context, common_options, aws_creds_options
+from samcli.commands._utils.options import (
+    metadata_override_option,
+    _TEMPLATE_OPTION_DEFAULT_VALUE,
+    get_or_default_template_file_name,
+)
+from samcli.commands._utils.resources import resources_generator
+from samcli.lib.telemetry.metrics import track_command
+
+SHORT_HELP = "Package an AWS SAM application."
+
+
+def resources_and_properties_help_string():
+    """
+    Total list of resources and their property locations that are supported for `sam package`
+    :return: str
+    """
+    return "".join(
+        f"\nResource : {resource} | Location : {location}\n".format(resource=resource, location=location)
+        for resource, location in resources_generator()
+    )
+
+
+HELP_TEXT = (
+    """The SAM package command creates a zip of your code and dependencies and uploads it to S3. The command
+returns a copy of your template, replacing references to local artifacts with the S3 location where the command
+uploaded the artifacts.
+
+The following resources and their property locations are supported.
+"""
+    + resources_and_properties_help_string()
+)
+
+
+@click.command("package", short_help=SHORT_HELP, help=HELP_TEXT, context_settings=dict(max_content_width=120))
+# TODO(TheSriram): Move to template_common_option across aws-sam-cli
```

Comment:
Why not just do it now?

Reply:
This will touch the full surface of commands across build, local etc. I'd rather have a separate PR for that.